azure-mgmt-batch/src/main/java/com/microsoft/azure/management/batch/implementation/ApplicationImpl.java

```diff
@@ -0,0 +1,154 @@
+            Application.UpdateDefinition<BatchAccount.Update>,
+            Application.Update {
+    private final ApplicationsInner client;
+
+    protected ApplicationImpl(String name, BatchAccountImpl batchAccount, ApplicationInner inner, ApplicationsInner client) {
+        super(name, batchAccount, inner);
+        this.client = client;
+    }
+
+    @Override
+    public String id() {
+        return this.inner().id();
+    }
+
+    @Override
+    public BatchAccountImpl parent() {
+        return this.parent;
+    }
+
+    @Override
+    public String displayName() {
+        return this.inner().displayName();
+    }
+
+    @Override
+    public List<ApplicationPackageInner> packages() {
+        return this.inner().packages();
+    }
+
+    @Override
+    public boolean allowUpdates() {
+        return this.inner().allowUpdates();
+    }
+
+    @Override
+    public String defaultVersion() {
+        return this.inner().defaultVersion();
+    }
+
+    @Override
+    public Observable<Application> createAsync() {
+        final ApplicationImpl self = this;
+        AddApplicationParametersInner createParameter = new AddApplicationParametersInner();
+        createParameter.withDisplayName(this.inner().displayName());
+        createParameter.withAllowUpdates(this.inner().allowUpdates());
+
+        return this.client.createAsync(this.parent.resourceGroupName(),
+                this.parent.name(),
+                this.name(), createParameter)
+                .map(new Func1<ApplicationInner, Application>() {
+                    @Override
+                    public Application call(ApplicationInner inner) {
+                        self.setInner(inner);
+                        return self;
+                    }
+                });
+    }
+
+    @Override
+    public Observable<Application> updateAsync() {
+        final ApplicationImpl self = this;
+
+        String applicationId = ResourceUtils.nameFromResourceId(this.inner().id());
+        UpdateApplicationParametersInner updateParameter = new UpdateApplicationParametersInner();
+        updateParameter.withDisplayName(this.inner().displayName());
+        updateParameter.withAllowUpdates(this.inner().allowUpdates());
+
+        return this.client.updateAsync(this.parent.resourceGroupName(),
+                this.parent.name(), applicationId, updateParameter)
+                .map(new Func1<Void, Application>() {
+                @Override
+                public Application call(Void result) {
+                    return self;
+                }
+        });
+    }
+
+    @Override
+    public Observable<Void> deleteAsync() {
+        return this.client.deleteAsync(this.parent.resourceGroupName(),
+                this.parent.name(),
+                this.name()).map(new Func1<Void, Void>() {
+            @Override
+            public Void call(Void result) {
+                return result;
+            }
```

Comment:
map is not required since client.deleteAsync returns `Observable<Void>` and ApplicationImpl::deleteAsync also returns `Observable<Void>`